Refactoring World Gen

brazyboi opened the pull request:
Refactored the terrain gen script so that it generates the noise onto a 2d list first, then places the tiles at the end. 

The ends of the worlds now "wrap"---though it's only terrain wise, the enemies do not cross the border with you. The idea later on is to just have a viewport, but this is good enough for now.  The player does in fact teleport to the other end of the world when it reaches one end.

brazyboi committed:
refactor terrain gen

legit the worst commit ever

brazyboi committed:
make it so the player teleports to the other side when they reach the end of the world

gavindg commented:
looping world is dope af would be legendary if you could make the thing that teleports the player a variable tho... maybe world size should be a global?

brazyboi committed:
smooth out the edges

brazyboi committed:
Changed the sides to be rendered by viewports

remaining issue is that the player's hitbox is smaller than the player itself, which makes it glitch on the border

brazyboi committed:
Made it so mining across the edge works

brazyboi committed:
fixed looping world

gavindg committed:
Merge branch 'main' into world-gen-modular

gavindg committed:
Revert "Merge branch 'main' into world-gen-modular"

This reverts commit 8dd60e8eb96ac6c5cfe9b6c4a6ea4700c0d5e10e, reversing
changes made to 78ab6981394c5f15630e475fe9a0d88386997916.
